subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -493,6 +493,24 @@ static int pa_sync_term_request(struct bt_conn *conn,
 	return err;
 }
 
+static bool bass_source_is_duplicate(uint32_t broadcast_id, uint8_t adv_sid, bt_addr_le_t addr)
+{
+	struct bass_recv_state_internal *state;
+
+	for (size_t i = 0; i < ARRAY_SIZE(scan_delegator.recv_states); i++) {
+		state = &scan_delegator.recv_states[i];
+
+		if (state != NULL && state->state.broadcast_id == broadcast_id
+		    && state->state.adv_sid == adv_sid && state->state.addr.type == addr.type) {
+			LOG_DBG("recv_state already exists at src_id=0x%02X", state->state.src_id);
+
+			return true;
```

Comment:
I think this may non-intuitive. Suggest to add reference to the BAP/BASS spec as why we are only checking these 3 values